trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/credentials/JsonIdentityProvider.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.aws.proxy.server.credentials;
+
+import com.fasterxml.jackson.databind.Module;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import io.trino.aws.proxy.spi.credentials.Identity;
+
+public class JsonIdentityProvider
+        implements Provider<Module>
```

Comment:
I think its worth fully quantifying `Module` so its clear it the jackson (not the guice) class.
```suggestion
        implements Provider<com.fasterxml.jackson.databind.Module>
```